R/downloads.R

```diff
@@ -115,8 +115,18 @@ hintr_prepare_agyw_download <- function(output, pjnz,
   assert_model_output_version(output, "2.7.16")
   progress <- new_simple_progress()
   progress$update_progress("PROGRESS_DOWNLOAD_AGYW")
-  dummy_data <- data.frame(x = c(1, 2, 3), y = c(3, 4, 5))
-  writexl::write_xlsx(list(sheet = dummy_data), path = path)
+
+  template_path <- naomi.resources::get_agyw_workbook_path()
+
+  risk_populations <- agyw_generate_risk_populations(output$model_output_path,
+                                                     pjnz)
+
+  sheets <- list(
+    "All outputs - F" = risk_populations$female_incidence,
+    "All outputs - M" = risk_populations$male_incidence,
+    "NAOMI outputs" = risk_populations$naomi_output
+  )
+  write_xlsx_sheets(template_path, sheets, path = path)
```

Comment:
What happens if something fails / throws an error in here?

Reply:
Thanks, I've written a wrapper which will add some more context should anything in excel writing error. Should return some more info to users https://github.com/mrc-ide/naomi/pull/413/commits/a10bfcd0a8f60407c69acf400e402f42736afb25#diff-8e11e9d56ce8b4fdd117a811bd64a4b2ea57306740475e9703d27faa1131cac3R1665